### PR TITLE
CONSOLE-3527: Create ManagedProxyServiceResolver for alertmanager-main services

### PR DESCRIPTION
**Pull request by TheRealJon**
Create a ManagedProxyServiceResolver that will allow the MCE cluster proxy to resolve requests to the alertmanager-main service on managed clusters.

[Required for openshift/console#12628](https://github.com/openshift/console/pull/12628)

### CODE REVIEW

File: pkg/console/controllers/advancedclustermanagement/managedproxyserviceresolver/controller.go [hunk 1]
```diff
@@ -96,34 +98,67 @@ func (c *ThanosQuerierProxyResolverController) Sync(ctx context.Context, control
...
 	if apierrors.IsNotFound(err) {
-		klog.V(4).Infoln("Skipping thanos-querier ManagedProxyServiceResolver sync. API is missing.")
+		klog.V(4).Infoln("Skipping ManagedProxyServiceResolver sync. API is not installed.")
 		return nil
 	}
 
 	if err != nil {
-		klog.V(4).Infof("Error getting ManagedProxyServiceResolver CRD: %s/n", err)
-		statusHandler.AddConditions(status.HandleProgressingOrDegraded(ThanosQuerierProxyResolverConditionPrefix, "GetManagedProxyServiceResolverCRDFailed", err))
+		klog.V(4).Infof("Error getting CustomrResourceDefinition for ManagedProxyServiceResolver: %v", err)
```
Comment (jhadvig) [inline pkg/console/controllers/advancedclustermanagement/managedproxyserviceresolver/controller.go:106]:
```suggestion
		klog.V(4).Infof("Error getting CustomResourceDefinition for ManagedProxyServiceResolver: %v", err)
```

File: pkg/console/subresource/managedproxyserviceresolver/apply.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func ApplyManagedProxyServiceResolver(ctx context.Context, client proxyclient.Ma
 
 		// Create failed, return error
 		if err != nil {
-			klog.V(4).Infof("failed to create ManagedServiceProxyResolver '%s': %s/n", required.Name, err)
+			klog.V(4).Infof("failed to create ManagedProxyServiceResolver '%s': %s/n", required.Name, err)
```
Comment (jhadvig) [inline pkg/console/subresource/managedproxyserviceresolver/apply.go:27]:
```suggestion
			klog.V(4).Infof("failed to create ManagedProxyServiceResolver %q: %v/n", required.Name, err)
```

File: pkg/console/subresource/managedproxyserviceresolver/apply.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func ApplyManagedProxyServiceResolver(ctx context.Context, client proxyclient.Ma
 
 	// Get failed, return error
 	if err != nil {
-		klog.V(4).Infoln("failed to get ManagedServiceProxyResolver '%s': %s/n", required.Name, err)
+		klog.V(4).Infoln("failed to get ManagedProxyServiceResolver '%s': %s/n", required.Name, err)
```
Comment (jhadvig) [inline pkg/console/subresource/managedproxyserviceresolver/apply.go:37]:
```suggestion
		klog.V(4).Infoln("failed to get ManagedProxyServiceResolver %q': %v/n", required.Name, err)
```

File: pkg/console/subresource/managedproxyserviceresolver/apply.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ func ApplyManagedProxyServiceResolver(ctx context.Context, client proxyclient.Ma
 
 		// Update failed, return error
 		if err != nil {
-			klog.V(4).Infof("failed to update ManagedServiceProxyResolver '%s': %s/n", required.Name, err)
+			klog.V(4).Infof("failed to update ManagedProxyServiceResolver '%s': %s/n", required.Name, err)
```
Comment (jhadvig) [inline pkg/console/subresource/managedproxyserviceresolver/apply.go:54]:
```suggestion
			klog.V(4).Infof("failed to update ManagedProxyServiceResolver %q: %v/n", required.Name, err)
```

**Review by jhadvig — COMMENTED** [review]
Just a few small nits. Otherwise looks good to go 👍

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Review by opayne1 — COMMENTED** [review]
/label docs-approved

**Comment by TheRealJon** [comment]
Closing this PR. The multicluster feature has been deferred.

**Pull request closed without merging**